File: src/main/java/alatoo/car_booking/services/admin/AdminService.java
package alatoo.car_booking.services.admin;

import alatoo.car_booking.dtos.BookACarDto;
import alatoo.car_booking.dtos.CarDto;
import alatoo.car_booking.dtos.CarDtoList;
import alatoo.car_booking.dtos.SearchCarDto;

import java.io.IOException;
import java.util.List;

public interface AdminService {
    boolean postCar(CarDto carDto);

    List<CarDto> getAllCars();

    void deleteCar(Long carId);

    CarDto getCarById(Long carId);

    boolean updateCar(Long carId, CarDto carDto) throws IOException;

    List<BookACarDto> getBookings();

    boolean changeBookingStatus(Long bookingId, String status);

    CarDtoList searchCar(SearchCarDto searchCarDto);
}

File: frontend/my-angular-app/src/app/modules/admin/services/admin.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { StorageService } from 'src/app/auth/services/storage/storage.service';

const BASIC_URL =["http://localhost:8081"];

@Injectable({
  providedIn: 'root'
})
export class AdminService {

  constructor(private http: HttpClient) { }

  postCar(carDto: any) {
    return this.http.post(BASIC_URL + "/api/admin/car", carDto, {
      headers: this.createAuthorizationHeader()
    });
  }

  getAllCars(): Observable<any> {
    return this.http.get(BASIC_URL + "/api/admin/cars", {
      headers: this.createAuthorizationHeader()
    })
  }

  deleteCar(carId: number): Observable<any> {
    return this.http.delete(BASIC_URL + "/api/admin/car/" + carId, {
      headers: this.createAuthorizationHeader()
    })
  }

  updateCar(carId: number, carDto: any) {
    return this.http.put(BASIC_URL + "/api/admin/car/" + carId, carDto, {
      headers: this.createAuthorizationHeader()
    });
  }

  getCarById(carId: number): Observable<any> {
    return this.http.get(BASIC_URL + "/api/admin/car/" + carId, {
      headers: this.createAuthorizationHeader(),
    });
  }


  createAuthorizationHeader(): HttpHeaders {
    let auhtHeaders: HttpHeaders = new HttpHeaders();
    return auhtHeaders.set('Authorization',
      'Bearer ' + StorageService.getToken()
    );
  }
  // createAuthorizationHeader(): HttpHeaders {
  //   const token = StorageService.getToken();
  //   if (!token) {
  //     console.error('Authorization token is missing!');
  //   }
  //   return new HttpHeaders().set('Authorization', 'Bearer ' + token);
  // }
  
}
